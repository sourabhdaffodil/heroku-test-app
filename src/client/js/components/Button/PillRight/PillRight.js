import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { media } from '../../../theme/style-utils';

const Pill = styled.div`
  border-top-right-radius: 3px;
  border-bottom-right-radius: 3px;
  padding: 3px 12px;
  background-color: ${props => props.theme.color('background', 'lighter')};

  ${media.phone`
    padding: 3px 5px;
  `}
`;

const PillRight = ({ children }) => (
  <Pill>
    { children }
  </Pill>
);

PillRight.propTypes = {
  children: PropTypes.any.isRequired,
};

export default PillRight;
