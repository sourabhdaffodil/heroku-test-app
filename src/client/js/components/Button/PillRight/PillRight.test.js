import React from 'react';
import { shallow } from 'enzyme';
import PillRight from './PillRight';

describe('Right pill button component', () => {
  it('Must render the left content of a pill button', () => {
    const wrapper = shallow(
      <PillRight>
        Left content
      </PillRight>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
