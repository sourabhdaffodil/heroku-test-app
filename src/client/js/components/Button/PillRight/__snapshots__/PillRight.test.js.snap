// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Right pill button component Must render the left content of a pill button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PillRight>
    Left content
  </PillRight>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": "Left content",
    },
    "ref": null,
    "rendered": "Left content",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "Left content",
      },
      "ref": null,
      "rendered": "Left content",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
